vote-sc/src/views.rs

```diff
@@ -0,0 +1,76 @@
+multiversx_sc::imports!();
+
+use crate::{
+    constants::{Hash, ProposalId, HASH_LENGTH, PROOF_LENGTH},
+    errors::NO_PROPOSAL,
+};
+
+#[multiversx_sc::module]
+pub trait ViewsModule {
+    #[view(getProposalRootHash)]
+    fn get_root_hash(
+        &self,
+        proposal_id: ProposalId,
+    ) -> OptionalValue<ManagedByteArray<HASH_LENGTH>> {
+        if self.root_hash_proposal_nonce(proposal_id).is_empty() {
+            return OptionalValue::None;
+        }
+
+        OptionalValue::Some(self.root_hash_proposal_nonce(proposal_id).get())
+    }
+
+    #[view(confirmVotingPower)]
+    fn confirm_voting_power(
+        &self,
+        proposal_id: ProposalId,
+        voting_power: BigUint<Self::Api>,
+        proof: ArrayVec<ManagedByteArray<HASH_LENGTH>, PROOF_LENGTH>,
+    ) -> bool {
+        match self.get_root_hash(proposal_id) {
+            OptionalValue::None => {
```

Comment:
I do not see the fallback mechanism to ID == 0 -> if roothash for proposal_id does not exist.